src/ig/IGExporter.ts

```diff
@@ -1172,71 +1003,39 @@ export class IGExporter {
-            `Since the "template" property is present in the ${this.configName}, an ig.ini file will be generated and ` +
-            `the ${filePathString} file will be ignored. NOTE: Support for the "template" property in ${this.configName} will ` +
-            `be removed in a future version of SUSHI. Please remove the "template" property in ${this.configName} and ` +
-            'manage the ig.ini file directly.';
-          logger.warn(message, { file: inputIniPath });
-        } else {
-          message +=
-            `The "template" property in ${this.configName} has been deprecated and will be ignored. The existing ` +
-            `${filePathString} file will be used instead.  To resolve this error, remove the "template" property in ` +
-            `${this.configName} and manage the ig.ini file directly.`;
-          logger.error(message, { file: inputIniPath });
-        }
+        const message =
+          `Found both a "template" property in ${this.configName} and an ig.ini file at ${filePathString}. ` +
+          `The "template" property in ${this.configName} is no longer supported and will be ignored. The existing ` +
+          `${filePathString} file will be used instead.  To resolve this error, remove the "template" property in ` +
+          `${this.configName} and manage the ig.ini file directly.`;
+        logger.error(message, { file: inputIniPath });
       } else {
-        let message =
-          `The "template" property in ${this.configName} has been deprecated. Please remove the "template" ` +
-          `property in ${this.configName} and manage the ig.ini file directly.`;
-        if (this.shouldCopyFiles) {
-          message +=
-            ` Support for the "template" property in ${this.configName} will be removed in a future ` +
-            'version of SUSHI.';
-          logger.warn(message);
-        } else {
-          message += ` To resolve this error, create an ig.ini file in your project folder with the following contents:\n\n${this.generateIgIniString(
-            false
-          )}`;
-          logger.error(message);
-        }
-      }
-
-      // Don't ever generate ig.ini in new IG Publisher project structure
-      if (!this.shouldCopyFiles) {
-        return;
+        const message =
+          `The "template" property in ${this.configName} is no longer supported. Please remove the "template" ` +
+          `property in ${this.configName} and manage the ig.ini file directly.` +
+          ` To resolve this error, create an ig.ini file in your project folder with the following contents:\n\n${this.generateIgIniString()}`;
+        logger.error(message);
       }
-
-      this.generateIgIni(igPath);
     } else if (existsSync(inputIniPath)) {
-      this.processIgIni(igPath, inputIniPath);
+      this.processIgIni(inputIniPath);
     } else {
-      // do nothing -- no template in config, no ig.ini in ig-data
+      // do nothing -- no template in config, no ig.ini provided
     }
```

Comment:
I have a question about this section of code. Previously, in the old tank set ups, we would generate an ig.ini file if the `template` property is used in configuration. If the template property is not used and there is an ig.ini file, we would process it and then copy it to the output directory.

With my updates, the `processIgIni` step only logs errors about missing required properties (an [IG] section and "ig" and "template" properties) and warnings for deprecated properties in the `ig.ini` file. However, this only happens if there is no `template` property in the config. Technically, there could be a case where someone has a `ig.ini` file and a `template` property, and we're missing logging these messages. This is true for the current version of SUSHI as well. However, since the `processIgIni` step is just informing people about properties in a file we do nothing with, I'm wondering if we just want to eliminate that step. Do others think there is value in providing these messages? Or should the whole `processIgIni` function be removed? That would mean that the `checkIgIni` function would just log an error about the presence of a "template" property in `sushi-config.yaml`.

Reply:
Oops this comment was supposed to be on [these lines](https://github.com/FHIR/sushi/pull/820/files#diff-20b28aa731fc636ee110f2e0e3b7f52fe153d4a10cc1f78f8c9299463cda71f2L1231-L1240).

Reply:
I think it's fine to keep our `ig.ini` warnings, they are helpful error messages in some cases I would guess. I also think the case where `template` is in the config and `ig.ini` is defined is fine, because in that case they get an error for having `template`, and once they remove it, we check their `ig.ini` for them.